GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -223,7 +223,8 @@ namespace gdjs {
      * @returns A float from 0 to 1.
      */
     getVolume(): float {
-      return this._volume;
+      if (this._id === null) return this._volume;
```

Comment:
Are you sure about this? Why do we have this._volume? Why is there a usage of it and a usage of this._howl.volume?

Reply:
this._volume is the volume given when we want to play a sound initially. That way, if the howl isn't loaded yet, we can keep it stored there until it is loaded, and so that we can set the volume correct volume on the howl when it starts playing. 

This is correct, because this._volume contains the future volume of the howl while it loads, and we can get the true howl volume on it directly once it is loaded. The this._volume will not be in sync since the fade function is on the howl and therefore messes with the internal howl volume directly.

That makes me think, is this kind of "automagical" function something we should be using? It is hard to tell what kind of overhead it has. Maybe an asynchronous action would be a good fit for doing that manually? Though, that would probably make the volume change more stuttering since it would be updating the volume at every game frame, not at a "sound update speed". I cannot tell if that would be a problem or not to be honest.

Reply:
> It is hard to tell what kind of overhead it has

You can check the source code of Howler: https://github.com/goldfire/howler.js/blob/143ae442386c7b42d91a007d0b1f1695528abe64/src/howler.core.js#L1295

If webaudio is used, it's using a native method (which is ideal).
otherwise, it's using setInterval. Not great (this could be handled by the game engine), but ok I guess and not too risky.